src/foam/dao/pg/SQLData.java

```diff
@@ -0,0 +1,144 @@
+package foam.dao.pg;
+
+import foam.core.FObject;
+import foam.core.PropertyInfo;
+
+import java.util.*;
+import java.util.ArrayList;
+import java.util.StringJoiner;
+
+/**
+ * prepares sql info based on a FObject
+ */
+public class SQLData {
```

Comment:
Why is this class needed? Can't this just be done in the Postgres DAO?

Reply:
This builds a collection of columnNames, which should be the same every time, so should be calculated in the DAO constructor rather than once per update, which is wasteful.
Also, copying the values out of the object itself before processing just generates extra work and garbage, especially for primitive values like numbers and booleans which will need to be promoted to real objects and then garbage collected.